balancer-js/src/modules/joins/joins.module.ts

```diff
@@ -302,53 +306,77 @@ export class Join {
     const poolId = node.id;
     const inputTokens: string[] = [];
     const inputAmts: string[] = [];
-    const childOutputRefs: string[] = []; // for testing purposes only
 
+    // inputTokens needs to include each asset even if it has 0 amount
     node.children.forEach((child) => {
-      if (child.outputReference !== '0') {
-        inputTokens.push(child.address);
-        inputAmts.push(
-          Relayer.toChainedReference(child.outputReference).toString()
-        );
-        childOutputRefs.push(child.outputReference);
-      }
+      inputTokens.push(child.address);
+      inputAmts.push(
+        child.outputReference !== '0'
+          ? Relayer.toChainedReference(child.outputReference).toString()
+          : '0'
+      );
     });
 
+    // TODO: check which types of pools should include their BPT in the input tokens
+    // TODO: check if isn't easier to move that logic to the graph
+    if (
+      node.type === PoolType.ComposableStable ||
+      node.type === PoolType.StablePhantom
+    ) {
```

Comment:
Which pool types should be includes on this check?
- I know that `AaveLinear` and `ERC4626Linear` also have BPT listed on tokensList, but they are joined through `batchSwap` instead of `joinPool`, so I believe they shouldn't be added.
- I added `StablePhantom` because its `joinAction` is set to `joinPool`.
Let me know if I should be able to find this documented somewhere.

Reply:
Not 100% sure on whats going on here. If its for handling non-leaf joins I think this should be handled elsewhere and we can focus on that in a later PR.

Reply:
Ok, so giving more context: `createJoinPool` was recently fixed by adding the BPT of the pool being joined to the `inputTokens` array because it needs to match what `getPoolTokens` returns, right?
If I understood correctly, `getPoolTokens` includes the pool BPT for some pool types (e.g. StablePhantom) and not for others (e.g. Weighted). This if/else intends to handle that logic.
Let me know if that's not the case.